code/mob/living/intangible/ai-camera.dm

```diff
@@ -282,6 +282,13 @@
 	resist()
 		return 0 //can't actually resist anything because there's nothing to resist, but maybe the hot key could be used for something?
 
+	//death stuff that should be passed to mainframe
+	gib(give_medal, include_ejectables) //this should be admin only, I would hope
+		message_admins("something gibbed the AI - if this wasn't an admin action, something has gone badly wrong")
+		return mainframe.gib(give_medal, include_ejectables)
```

Comment:
I'm not sure this is a good idea, there's a lot of M.gib() calls in the code, and the possibility that one of them is going to hit the eye and gib an AI for flying over something seems bad.

Reply:
There's a reason for this, and it's that I called gib on the player instance with the player panel and it caused the AI eye to spew gibs everywhere and get into a weird state.

Maybe we should just log it and return instead?